theme/type/responsiveType.less

```diff
@@ -9,13 +9,23 @@
   // Concatenate type hierarchy variable name
   @type-scale: "@{variant}-type-scale";
   @row-span: "@{variant}-type-row-span";
+  @weight: "@{variant}-type-weight";
   @mobile-type-scale: "@{type-scale}-mobile";
   @mobile-row-span: "@{row-span}-mobile";
+  @mobile-weight: "@{variant}-type-weight-mobile";
 
   @media only screen and (max-width: @max-breakpoint) {
     .__baselineType(@@mobile-type-scale, @@mobile-row-span);
+
+    & when (not(@@mobile-weight = 400)) {
```

Comment:
Favour simpler code IMO.  Setting `400` to `400` is harmless so may as well save the developer brain cycles, `LESS` won't care :)